### PR TITLE
Add C implementation for `@stdlib/math/base/special/sqrtpi`

### LINKED ISSUE #676: [RFC]: Add C implementation for `@stdlib/math/base/special/sqrtpi`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/sqrtpi`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/sqrtpi)
```c
double stdlib_base_sqrtpi( const double x ) 
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #676 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks passed, this pr can have a final review.

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md [hunk 1]
```diff
@@ -82,6 +82,97 @@ for ( i = 0; i < 100; i++ ) {
...
+
+Computes the principal [square root][@stdlib/math/base/special/sqrt] of the product of π and a positive number.
+
+```c
+double x = stdlib_base_sqrtpi( 4.0 );
+// returns ~3.5449
+
+x = stdlib_base_sqrtpi( 10.0 );
+// returns ~5.60499
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md [hunk 2]
```diff
@@ -82,6 +82,97 @@ for ( i = 0; i < 100; i++ ) {
...
+int main() {
+    double x;
+    double v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+        x = (double)rand() / (double)RAND_MAX * 100.0;
+        v = stdlib_base_sqrtpi( x );
+        printf( "sqrtpi(%f) = %f\n", x, v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md]:
```suggestion
        printf( "sqrtpi(%lf) = %lf\n", x, v );
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md [hunk 3]
```diff
@@ -82,6 +82,97 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+        x = (double)rand() / (double)RAND_MAX * 100.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/README.md]:
```suggestion
        x = ( (double)rand() / (double)RAND_MAX ) * 100.0;
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+*
+* @return clock time
+*/
+double tic() {
+	struct timeval now;
+	gettimeofday( &now, NULL );
+	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
+}
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/benchmark/c/benchmark.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+double benchmark() {
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( (double) rand() * 100000.0 ) - 0.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/benchmark/c/benchmark.c]:
```suggestion
		x = ( (double)rand() * 100000.0 ) - 0.0;
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/examples/c/example.c [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+        x = (double)rand() / (double)RAND_MAX * 100.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/examples/c/example.c]:
```suggestion
        x = ( (double)rand() / (double)RAND_MAX ) * 100.0;
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        v = stdlib_base_sqrtpi( x );
+        printf( "sqrtpi(%f) = %f\n", x, v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/examples/c/example.c]:
```suggestion
        printf( "sqrtpi(%lf) = %lf\n", x, v );
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/include/stdlib/math/base/special/sqrtpi.h [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef STDLIB_MATH_BASE_SPECIAL_SQRTPI_H
+#define STDLIB_MATH_BASE_SPECIAL_SQRTPI_H
+
+#include <stdint.h>
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/include/stdlib/math/base/special/sqrtpi.h]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/lib/native.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Computes the principal square root of the product of 𝛑 and a positive number.
+*
+* @param {number} x - input value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/lib/native.js:32]:
```suggestion
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/src/main.c [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+#include "stdlib/math/base/special/sqrtpi.h"
+#include "stdlib/math/base/special/sqrt.h"
+#include "stdlib/constants/float64/pi.h"
+
+/**
+* Compute the principal square root of the product of π and a positive number.
+*
+* @param x    input value.
+* @return	  output value.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/src/main.c]:
```suggestion
* @param x    input value
* @return	  output value
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpi/src/main.c [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+*
+* @example
+* double x = 4.0;
+*
+* double out = stdlib_base_sqrtpi( x );
+* // returns 3.5449
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sqrtpi/src/main.c]:
```suggestion
* double out = stdlib_base_sqrtpi( 4.0 );
* // returns 3.5449
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**